commands/post_test.go

```diff
@@ -0,0 +1,106 @@
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestPostCreateCmdF() {
+	s.Run("create a post with empty text", func() {
+		cmd := &cobra.Command{}
+
+		err := postCreateCmdF(s.client, cmd, []string{"some-channel", ""})
+		s.Require().NotNil(err)
+	})
+
+	s.Run("no channel specified", func() {
+		msgArg := "some text"
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("message", msgArg, "")
+
+		err := postCreateCmdF(s.client, cmd, []string{"", msgArg})
+		s.Require().NotNil(err)
+	})
+
+	s.Run("wrong reply msg", func() {
+		msgArg := "some text"
+		replyToArg := "a-non-existing-post"
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("message", msgArg, "")
+		cmd.Flags().String("reply-to", replyToArg, "")
+
+		s.client.
+			EXPECT().
+			GetPost(replyToArg, "").
+			Return(nil, &model.Response{Error: &model.AppError{Message: "some-error"}}).
+			Times(1)
+
+		err := postCreateCmdF(s.client, cmd, []string{msgArg})
+		s.Require().NotNil(err)
```

Comment:
Same `EqualError` change here

Reply:
Since the implementation of `Error()` in AppError formats the error message:
```golang
func (er *AppError) Error() string {
    return er.Where + ": " + er.Message + ", " + er.DetailedError
}
```
I suggest that we can use error check like this:
```golang
s.Require().NotNil(err) // this can be omitted
s.Require().Contains(err.Error(), "some-error")
```